cmake/thirdparty/SetupOpenMP.cmake

```diff
@@ -52,7 +52,18 @@ endif()
 message(STATUS "OpenMP Compile Flags: ${_compile_flags}")
 message(STATUS "OpenMP Link Flags:    ${_link_flags}")
 
-blt_import_library(NAME openmp
+# Don't create the openmp target if it already exists.
+if (NOT TARGET blt::openmp)
+    blt_import_library(NAME openmp
```

Comment:
Do we need to move the location of the blt_import_library into a location that will get installed along with these files?

Reply:
The macros will get installed alongside the TPL .cmake files (i.e. `SetupCUDA.cmake` etc).  See https://github.com/LLNL/blt/blob/1e14a6ee5c001e10ce2e64efd8f99b313b49a80f/cmake/BLTMacros.cmake#L1546

Reply:
Do we want to install all of our macros or just the `blt_import_library` macro?

Reply:
I think we need to install all the macros due to the dependence of the `SetupMPI` etc on `blt_list_remove_duplicates` and other macros.  We could split up our macros more so that we only install a few of the required ones.

Reply:
I am concerned that installing all of `BLTMacros.cmake` implies that those macros should work in the downstream projects w/o the whole of BLT. This will not actually be true because a lot of these macros have internal macros in `BLTPrivateMacros.cmake` and also rely on other variables set in other locations. For example, `blt_append_custom_compiler_flag` needs all the `*_COMPILER_FAMILY_IS_*` set in `SetupCompilerOptions.cmake`.

Reply:
I agree that installing all of `BLTMacros.cmake` without guaranteeing they work in all downstream projects isn't ideal.  However, `BLTPrivateMacros.cmake` will be install so in this case resolving the macro names should at least work.  Many variables will not be set

Reply:
My opinion: The CMake exported target logic source files we install  (.cmake ones) should be as simple as possible and should not relay on BLT macros. 
With newer cmake, its better to use the bona-fide cmake targets for OpenMP and MPI, if at all possible the BLT targets should be come an alias or pass though to those (but that may be a step beyond what is intended for this PR)

Reply:
I agree that's a good goal, maybe slightly premature. Moving to this approach for tpls will help simplify things, e.g. a lot of cruft setting up HIP was so that we could "re export" the targets (which will no longer be necessary)

Reply:
I agree we should install a minimal amount of code, but in this case we would need to repeat the internals of `blt_import_target` as well as `blt_list_remove_duplicates` and other dependencies within each `Setup*.cmake` file.  We could break off a smaller, installable set of macros `BLTMacros` into a separate file.  Thoughts?

As for moving to native cmake targets, I'd like to handle some of that work in a follow-up PR.